clang/lib/CIR/CodeGen/CIRGenAsm.cpp

```diff
@@ -26,6 +28,98 @@ static AsmFlavor inferFlavor(const CIRGenModule &cgm, const AsmStmt &s) {
   return isa<MSAsmStmt>(&s) ? AsmFlavor::x86_intel : gnuAsmFlavor;
 }
 
+// FIXME(cir): This should be a common helper between CIRGen
+// and traditional CodeGen
+static std::string simplifyConstraint(
+    const char *constraint, const TargetInfo &target,
+    SmallVectorImpl<TargetInfo::ConstraintInfo> *outCons = nullptr) {
+  std::string result;
+
+  while (*constraint) {
+    switch (*constraint) {
+    default:
+      result += target.convertConstraint(constraint);
+      break;
+    // Ignore these
+    case '*':
+    case '?':
+    case '!':
+    case '=': // Will see this and the following in mult-alt constraints.
+    case '+':
+      break;
+    case '#': // Ignore the rest of the constraint alternative.
+      while (constraint[1] && constraint[1] != ',')
+        constraint++;
+      break;
+    case '&':
+    case '%':
+      result += *constraint;
+      while (constraint[1] && constraint[1] == *constraint)
+        constraint++;
+      break;
+    case ',':
+      result += "|";
+      break;
+    case 'g':
+      result += "imr";
+      break;
+    case '[': {
+      assert(outCons &&
+             "Must pass output names to constraints with a symbolic name");
+      unsigned index;
+      bool resolveResult =
+          target.resolveSymbolicName(constraint, *outCons, index);
+      assert(resolveResult && "Could not resolve symbolic name");
+      (void)resolveResult;
+      result += llvm::utostr(index);
+      break;
+    }
+    }
+
+    constraint++;
+  }
+
+  return result;
+}
+
+// FIXME(cir): This should be a common helper between CIRGen
+// and traditional CodeGen
```

Comment:
Please pick one comment style for the whole block of comments, either `//` or `///`